1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:
   Region A: [10, 15, 12, 8, 14]
   Region B: [18, 20, 16, 22, 25]
   Calculate the mean sales for each region.


In [1]:
A = [10, 15, 12, 8, 14] 
B = [18, 20, 16, 22, 25]

https://docs.google.com/document/d/1Ioc-F8oiwmt_afbvjEXvSc6Dk5bBI2dzZSnbn9fCWbo/edit

In [5]:
import numpy as np

In [3]:
np.mean(A)

11.8

In [4]:
np.mean(B)

20.2

2. Scenario: A survey is conducted to measure customer satisfaction on a scale of 1 to 5. The data collected is as follows:
   [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
   Calculate the mode of the survey responses.


In [17]:
A = [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]

In [20]:
A1 = np.array(A)

In [21]:
from scipy import stats

In [30]:
stats.mode(A1)

C:\Users\HP\AppData\Local\Temp\ipykernel_12852\3260776875.py:1: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  stats.mode(A1)


ModeResult(mode=array([4]), count=array([3]))

3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:
   Department A: [5000, 6000, 5500, 7000]
   Department B: [4500, 5500, 5800, 6000, 5200]
   Calculate the median salary for each department.


In [32]:
A = [5000, 6000, 5500, 7000]
B = [4500, 5500, 5800, 6000, 5200]

In [35]:
print(np.median(A))
print(np.median(B))

5750.0
5500.0


4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:
   [25.5, 24.8, 26.1, 25.3, 24.9]
   Calculate the range of the stock prices.


5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:
   Group A: [85, 90, 92, 88, 91]
   Group B: [82, 88, 90, 86, 87]
   Perform a t-test to determine if there is a significant difference in the mean scores between the two groups.


In [7]:
A = [85, 90, 92, 88, 91]

In [8]:
B = [82, 88, 90, 86, 87]

In [9]:
stats.ttest_ind(a=A, b=B, equal_var=True)

Ttest_indResult(statistic=1.4312528946642733, pvalue=0.19023970239078333)

Here, since the p-value (0.1902) is greater than alpha = 0.05 so we cannot reject the null hypothesis of the test. There is a significant difference in the mean scores between the two groups.

6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
   Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
   Sales (in thousands): [25, 30, 28, 20, 26]
   Calculate the correlation coefficient between advertising expenditure and sales.


In [2]:
Advertising_Expenditure = [10, 15, 12, 8, 14]

In [4]:
Sales = [25, 30, 28, 20, 26]

In [10]:
import pandas as pd

In [13]:
df = pd.DataFrame({"Advertising_Expenditure":Advertising_Expenditure, "Sales":Sales})

In [14]:
df.corr()

Advertising_Expenditure     Sales
Advertising_Expenditure                 1.000000  0.875751
Sales                                   0.875751  1.000000

In [26]:
from scipy import stats 
stats.pearsonr(Advertising_Expenditure, Sales)

PearsonRResult(statistic=0.8757511375750133, pvalue=0.051583194188217156)

In [27]:
stats.spearmanr(Advertising_Expenditure, Sales)

SignificanceResult(statistic=0.8999999999999998, pvalue=0.03738607346849874)

7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:
   [160, 170, 165, 155, 175, 180, 170]
   Calculate the standard deviation of the heights.


In [37]:
A = [160, 170, 165, 155, 175, 180, 170]

In [38]:
np.std(A)

7.953949089757174

8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:
   Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]
   Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]
   Perform a linear regression analysis to predict job satisfaction based on employee tenure.


In [39]:
import pandas as pd

In [42]:
A = pd.Series([2, 3, 5, 4, 6, 2, 4])

In [43]:
B = pd.Series([7, 8, 6, 9, 5, 7, 6])

In [59]:
X = pd.DataFrame({"Year":A, "Job_Satisfaction": B})

In [60]:
X1 = X["Year"]

In [83]:
X1 = pd.DataFrame(X1)

In [85]:
X1.shape

(7, 1)

In [86]:
y = X["Job_Satisfaction"]

In [87]:
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split

In [88]:
X_train, X_test, y_train, y_test = train_test_split(X1, y)

In [89]:
lr = LinearRegression()

In [90]:
X_train = pd.DataFrame(X_train)

In [91]:
X_train.shape

(5, 1)

In [92]:
lr.fit(X_train,y_train)

LinearRegression()

In [93]:
X

Year  Job_Satisfaction
0     2                 7
1     3                 8
2     5                 6
3     4                 9
4     6                 5
5     2                 7
6     4                 6

In [96]:
y_predict = lr.predict(X_test)

In [97]:
y_test

3    9
1    8
Name: Job_Satisfaction, dtype: int64

In [98]:
y_predict

array([6.109375, 6.5625  ])

In [100]:
from sklearn.metrics import r2_score
from sklearn.metrics import mean_squared_error

In [101]:
r2_score(y_test, y_predict)

-19.84423828125

9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:
   Medication A: [10, 12, 14, 11, 13]
   Medication B: [15, 17, 16, 14, 18]
   Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.


In [1]:
from scipy.stats import f_oneway

In [2]:
A = [10, 12, 14, 11, 13]

In [3]:
B = [15, 17, 16, 14, 18] 

In [4]:
f_oneway(A, B)


F_onewayResult(statistic=16.0, pvalue=0.003949772803445326)

10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is

 as follows:
    [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
    Calculate the 75th percentile of the feedback ratings.


In [103]:
A =  [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]

In [104]:
np.percentile(A,75)

8.75

11. Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:
    [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
    Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.


In [15]:
weights = [10.2, 9.8, 10.0, 10.5, 10.3, 10.1] 

H0 = 10; 
H1 ≠ 10

In [28]:
from scipy.stats import ttest_1samp  
import numpy as np  

In [30]:
t_test, p_val = ttest_1samp(weights, 10)

In [31]:
p_val

0.19077595151110102

In [ ]:
p_val>0.05 we fail to reject null hypothesis that means  mean weight does not differs from 10 grams.

12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:
    Design A: [100, 120, 110, 90, 95]
    Design B: [80, 85, 90, 95, 100]
    Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs.


In [34]:
from scipy.stats import chi2_contingency


13. Scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:
    [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
    Calculate the 95% confidence interval for the population mean satisfaction score.


In [1]:
import numpy as np
import scipy.stats as st

In [2]:
data = [7, 9, 6, 8, 10, 7, 8, 9, 7, 8] 

In [9]:
# create 95% confidence interval
st.t.interval(alpha=0.95,
              df=len(data)-1,
              loc=np.mean(data), 
              scale=st.sem(data))

C:\Users\HP\AppData\Local\Temp\ipykernel_14916\2382268529.py:2: DeprecationWarning: Use of keyword argument 'alpha' for method 'interval' is deprecated and wil be removed in SciPy 1.11.0. Use first positional argument or keyword argument 'confidence' instead.
  st.t.interval(alpha=0.95,


(7.043561120599888, 8.756438879400113)

14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:
    Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
    Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]
    Perform a simple linear regression to predict performance based on temperature.


In [21]:
temp =  [20, 22, 23, 19, 21] 
Performance = [8, 7, 9, 6, 8]

In [22]:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
import pandas as pd


In [23]:
df1 = pd.DataFrame({"temp":temp,"Performance":Performance })

In [24]:
df1

temp  Performance
0    20            8
1    22            7
2    23            9
3    19            6
4    21            8

In [26]:
X = df1[["temp"]]

In [27]:
y = df1[["Performance"]]

In [32]:
X_train

temp
0    20
3    19
4    21
1    22

In [33]:
 X_test

temp
2    23

In [28]:
X_train, X_test, y_train, y_test = train_test_split(X, y,test_size = 0.2)

In [29]:
lr = LinearRegression().fit(X_train,y_train )

In [31]:
lr.predict(X_test)

array([[8.]])

15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:
    Group A: [4, 3, 5, 2, 4]
    Group B: [3, 2, 4, 3, 3]
    Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.


16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:
    [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
    Calculate the interquartile range (IQR) of the ages.


In [34]:
import numpy as np
  
data = [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
  
# First quartile (Q1)
Q1 = np.percentile(data, 25)
  
# Third quartile (Q3)
Q3 = np.percentile(data, 75)
  
# Interquaritle range (IQR)
IQR = Q3 - Q1
  
print(IQR)

22.5


17. Scenario: A study is conducted to compare the performance of three different machine learning algorithms. The accuracy scores for each algorithm are as follows:
    Algorithm A: [0.85, 0.80, 0.82, 0.87, 0.83]
    Algorithm B: [0.78, 0.82, 0.84, 0.80, 0.79]
    Algorithm C: [0.90, 0.88, 0.89, 0.86, 0.87]
    Perform a Kruskal-Wallis test to determine if there is a significant difference in the median accuracy scores between the algorithms.


18. Scenario: A company wants to analyze the effect of price on sales. The data collected is as follows:
    Price (in dollars): [10, 15, 12, 8, 14]
    Sales: [100, 80, 90, 110, 95]
    Perform a simple linear regression to predict

 sales based on price.


In [35]:
Price = [10, 15, 12, 8, 14] 
Sales =  [100, 80, 90, 110, 95]
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
import pandas as pd
df1 = pd.DataFrame({"Price":Price,"Sales":Sales })
X = df1[["Price"]]
y = df1[["Sales"]]
X_train, X_test, y_train, y_test = train_test_split(X, y,test_size = 0.2)
lr = LinearRegression().fit(X_train,y_train )
lr.predict(X_test)

array([[83.17757009]])

19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:
    [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
    Calculate the standard error of the mean satisfaction score.


In [42]:
from scipy.stats import sem

#define dataset 
data =[7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
#calculate standard error of the mean 
sem(data)



0.30550504633038933

20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
    Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
    Sales (in thousands): [25, 30, 28, 20, 26]
    Perform a multiple regression analysis to predict sales based on advertising expenditure.


In [44]:
Expenditure = [10, 15, 12, 8, 14]
Sales =  [100, 80, 90, 110, 95]
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
import pandas as pd
df1 = pd.DataFrame({"Expenditure":Expenditure,"Sales":Sales })
X = df1[["Expenditure"]]
y = df1[["Sales"]]
X_train, X_test, y_train, y_test = train_test_split(X, y,test_size = 0.2)
lr = LinearRegression().fit(X_train,y_train )
lr.predict(X_test)

array([[83.17757009]])